### DISCUSSION TITLE
Batch jpg mogrify job leaves xxxx.jpg~ temporary/backup files behind....

### DISCUSSION THREAD

**Discussion by Montreal666** [ImageMagick/ImageMagick#2754] [General]
Runing a batch jpg mogrify job which leaves multiple xxxx.jpg~ temporary/backup files behind....
with one or multiple "tilde" symbols after the file extension.
Folder contains about 10 000 files.
searched google all over and can't find an explanation.

And yes I don't mind using the same folder/overwriting files.

command is:
**magick mogrify -fuzz 1% -trim +repage -resize 600^ -gravity Center -crop 600x600+0+0 *.jpg** 
And seems to work as intended.

![image](https://user-images.githubusercontent.com/56667580/96927724-7c16b180-1485-11eb-83d8-30629ac0c1ad.png)

Is this Imagemagick related or some other application messing around?
If by design, is it normal that these files remain after the job is completed?
I didn't see any error report log so I assume everything worked out (or is there some log I can check ?)

looking for guidance before I write yet another script to delete these files automatically.

Thx,

**Comment by fmw42** [comment]
In what directory are you running the command? What permissions on the directory?  I do not recommend using either the install directory or the /tmp directory for your work. Create a new directory in your User home directory.  How much space do you have left in your /tmp directory?  Is it pretty full? Can you clean it out?  Does it have sufficient disk space?

**Comment by snibgo** [comment]
What version of IM, on what platform?

IM should remove each tilde ~ file, but does not check that this has succeeded because a failure won't affect the success of the overall operation. 

So this might happen because your OS doesn't permit IM to delete files from that directory.

**Reply by Montreal666** [reply]
lauching the command directly in the intended directory, no tmp. not needed as I don't care about the original files.

> So this might happen because your OS doesn't permit IM to delete files from that directory.

makes sense, probably launching the command line without admin rights.... trying again

Does IM work on multiple files at the same time or if I start seeing multiple tilde files it means IM can't delete them?

Thx to both of you.

**Reply by fmw42** [reply]
What are the permissions of the /tmp directory or where ever you are seeing the ~ files?

**Comment by snibgo** [comment]
I repeat: What version of IM, on what platform?

> Does IM work on multiple files at the same time or if I start seeing multiple tilde files it means IM can't delete them?

Well, yes, any tilde files remaining when the command is finished have not been successfully deleted.

EDIT to add: I suspect a permissions problem, either on the input files or the directory.

**Comment by Montreal666** [comment]
Windows 10. Just downloaded IM so latest version. Also suspecting permissions issue. Changed from admistrator to super administrator to avoid UAC and permissions issues and/or will try to ensure I have elevated privileges set accordingly. Didn't have time to retry the batch job. Will check back tomorrow. Thx for the feedback.

**Comment by snibgo** [comment]
For Windows, I suggest the following commands:
```
attrib "A Ha - Take On Me.jpg~"

icacls "A Ha - Take On Me.jpg~"
```
If "attrib" shows the R (read-only) attribute is set, then that is the problem. Use "attrib -R {filename}" to unset it.

icacls is more complex, but check if your username has full access ("F"). If you don't, then use "icacls {filename} /grant {username}:F".

Ideally, do this on the directory before running the mogrify command, so before the input files are renamed with a tilde.

(The equivalent bash command is "chmod".)

**Comment by Montreal666** [comment]
Got it working; the folder I was using was created automatically by an Android syncing program (SyncMe), 
therefore the folder owner in advanced security settings was set to "guest";
changed back to administrator and all is good now; thanks for your assistance.

**Comment by kurema** [comment]
In my case, "mogrify -format webp -crop 10x10+0+0 *.png *.webp" when there's no png file caused this issue. It worked fine without *.png.
Environment: Ubuntu 20.04.6 LTS on WSL2 on Windows 11.
